### ISSUE TITLE
Nested Container launch from inside outer level Container FAILS.

### ISSUE THREAD

**Issue by ColumGaynor** [lxc/lxd#4688]
```
# Required information

 * Distribution:                                               Ubuntu
 * Distribution version:                                  16.10 Yakkety Yak
 * The output of "lxc info" or if that fails:
gaynor@gaynor-Aspire-E5-521:~$ lxc info
config: {}
api_extensions:
- storage_zfs_remove_snapshots
- container_host_shutdown_timeout
- container_stop_priority
- container_syscall_filtering
- auth_pki
- container_last_used_at
- etag
- patch
- usb_devices
- https_allowed_credentials
- image_compression_algorithm
- directory_manipulation
- container_cpu_time
- storage_zfs_use_refquota
- storage_lvm_mount_options
- network
- profile_usedby
- container_push
- container_exec_recording
- certificate_update
- container_exec_signal_handling
- gpu_devices
- container_image_properties
- migration_progress
- id_map
- network_firewall_filtering
- network_routes
- storage
- file_delete
- file_append
- network_dhcp_expiry
- storage_lvm_vg_rename
- storage_lvm_thinpool_rename
- network_vlan
- image_create_aliases
- container_stateless_copy
- container_only_migration
- storage_zfs_clone_copy
- unix_device_rename
- storage_lvm_use_thinpool
- storage_rsync_bwlimit
- network_vxlan_interface
- storage_btrfs_mount_options
- entity_description
- image_force_refresh
- storage_lvm_lv_resizing
- id_map_base
- file_symlinks
- container_push_target
- network_vlan_physical
- storage_images_delete
- container_edit_metadata
- container_snapshot_stateful_migration
- storage_driver_ceph
- storage_ceph_user_name
- resource_limits
- storage_volatile_initial_source
- storage_ceph_force_osd_reuse
- storage_block_filesystem_btrfs
- resources
- kernel_limits
- storage_api_volume_rename
- macaroon_authentication
- network_sriov
- console
- restrict_devlxd
- migration_pre_copy
- infiniband
- maas_network
- devlxd_events
- proxy
- network_dhcp_gateway
- file_get_symlink
- network_leases
- unix_device_hotplug
- storage_api_local_volume_handling
- operation_description
- clustering
- event_lifecycle
- storage_api_remote_volume_handling
- nvidia_runtime
- container_mount_propagation
- container_backup
api_status: stable
api_version: "1.0"
auth: trusted
public: false
auth_methods:
- tls
environment:
  addresses: []
  architectures:
  - x86_64
  - i686
  certificate: |
    -----BEGIN CERTIFICATE-----
    MIIFdjCCA16gAwIBAgIQNzxkNayqRBGUZcrE/bQPtzANBgkqhkiG9w0BAQsFADBC
    MRwwGgYDVQQKExNsaW51eGNvbnRhaW5lcnMub3JnMSIwIAYDVQQDDBlyb290QGdh
    eW5vci1Bc3BpcmUtRTUtNTIxMB4XDTE4MDYyMjIwMDgwN1oXDTI4MDYxOTIwMDgw
    N1owQjEcMBoGA1UEChMTbGludXhjb250YWluZXJzLm9yZzEiMCAGA1UEAwwZcm9v
    dEBnYXlub3ItQXNwaXJlLUU1LTUyMTCCAiIwDQYJKoZIhvcNAQEBBQADggIPADCC
    AgoCggIBALvrS9wvpGb0zKjXy0JKsQn8l2fScFDewzhzC8f8Ux4Tf9UrraCh9NSf
    CG62gjs29GzcXDXxiXiGPIV8Al3wP7Z6E+JNyP55tckfP6RkW5f1+/y4L+WaCCXu
    oYyjClatogSjVUVkGT5HBEIspXouuxDCEztKiyh6+3wpaCW2Jl/QkBDjHDUhle8L
    jP+yXA9oybnNLuW6C9VTPS+7vDL1IGjcaZj6r+9X5BOp6W0Pg16g5CJZWphzmrgs
    20cJ2btf7Zn1uqz9G8CfJgZLJgo7XHZuXtquM5bhUJKhQqEFWmymtlGpmFk1H3Hl
    qQtInsA+zQvhrV5WM5Uj5ltm15AuZTQqKUjDC0c8RnziroGzixNRPpd7oX4+16PI
    cvnYGfmKgNMxlir0zepgMKb0pQkfoPWvAt/ZpUj2vb7Mao2wlKldw1ubXhu1jDqZ
    kcz1xDXi/PTYYhm8547w1W1LyiZaK82ryFO77PnP30n7spexz3FuNV8cHBqqr/Kb
    ZU7aMN9x/9L73IpDRx9OTqtmbwc0ftL3RPOl+2HnmF443P0ULIFHO43ZDwaP+Z0D
    lo7CP8O0vDLwDeXQc6q5L5GYwI2u2++vDyv2g3hdJ0Yuy8G2zOaJJwewFGi+LreZ
    kjn3NFBQDT3qPXPHwo7pTe0Ix/JahgJwWcmuiUakvrS3lpPMTUiLAgMBAAGjaDBm
    MA4GA1UdDwEB/wQEAwIFoDATBgNVHSUEDDAKBggrBgEFBQcDATAMBgNVHRMBAf8E
    AjAAMDEGA1UdEQQqMCiCFGdheW5vci1Bc3BpcmUtRTUtNTIxhwTAqP87hwSsEQAB
    hwTAqHoBMA0GCSqGSIb3DQEBCwUAA4ICAQAhPJpMIbzupT4qCP1qnmiZvGNQSDJc
    E7lNCROfpREk2k/NdsQuNXU6CHyadb0avQ59C/dMiPZLZnld9D9jfIV41XQPIn4c
    i/VxTkWgk4IDJjNYbkx8IviK29NZ6jwfKJUBvEvixpX6clsoL5D4kmEZ6LIh/q1d
    rygPDDGpTgP3IuM4/qYzpzUfT8WlrqE1noMCPBNWkpu3K0DFinCDP0+/VFRT1p3F
    j/FCW4TxjlrNHv9PLLNBt0AtZnWnatgO+Ai5raWSptIg4Netr2QfUcgftN0Lgm+K
    Kma0lq0XSOI0o2NdQ4/H89SaK/hVTgpLYx0eUuJ9wRYMwHWPsrZORGVnFU2vlcch
    8y+rrslVEOX5KY1JgAXUQ/quaDgLgxOgCdjZadeozUQf4qHe3BXQ+aoFfsP9TaI5
    /OAhCtCYF3dEbQSzVZLSvgYlo7e0afnEIpezz5bRyuyJWraBWsrYIncOjt6vW6ZY
    0R39/iYxkwWAT9P98Y8A9hzvi5bz2CAw7bUA/VAyOyFrMcmpt9W6B8T57vZ0Jf1X
    XuT+ZnJnFnnEpbqyoiDfimvYu8mF6P+i8iuqpL+tMMBM2U3B/7A8Fd0D6xw0ZGAF
    C/UvfpCvxcPvC6WhVYin3DiFdEI0jdlEGRojMrU/F7ESDR6jI8QTpZvS16ggTsQH
    FR6kG6Guy4ajSg==
    -----END CERTIFICATE-----
  certificate_fingerprint: cce4d733d80c4335b60e680eaccfcbd1850e9967789be791ed326e4e2eac6c18
  driver: lxc
  driver_version: 3.0.1
  kernel: Linux
  kernel_architecture: x86_64
  kernel_version: 4.8.0-59-generic
  server: lxd
  server_pid: 6835
  server_version: "3.1"
  storage: dir
  storage_version: "1"
  server_clustered: false
  server_name: gaynor-Aspire-E5-521


# Issue description

Attempt to launch an inner LXD container from within an outer LXD container FAILS!

# Steps to reproduce

1. Bare-metal Host machine (Host name = gaynor-Aspire-E5-521)is running Ubuntu 16.10 desktop (Yakkety Yak).
2. Used snap to install LXD and ran the sudo lxd init   Storage backend = dir and standard bridge lxdbr0 in use.
3. From the Yakkety Host machine, now launched an LXD machine container using the standard Ubuntu 16.04 (Xenial) image.
   Result: Success - Container named "informed-impala" is successfully launched.
4. From Yakkety Host machine can successfully take a session inside the "informed-impala" container [ lxc exec inform-impala bash ]  
5. From the Yakkety Host machine, now set up "security.nesting true" configuration setting for the "informed-impala" (Xenial) container.
6. From the "informed-impala" (Xenial) container tried to launch an inner-container based on ArchLinux image from "images:" remote repository
   Result: OK - The image is downloaded into the image cache of the "informed-impala" (Xenial Container) 
           OK - The Arch Linux image is stored inside the image cach of LXD machine container "informed-impala" (Xenial)
 		   OK - A new LXD (inner) Machine container named "nice-malamute" is created inside the "informed.impala"(Xenial) container, however...
           NOT-OK - The launch of the inner container FAILS!

<<<Console Sequence Capture>>>
 
gaynor@gaynor-Aspire-E5-521:~$ lxc image list
+-------+--------------+--------+---------------------------------------------+--------+----------+------------------------------+
| ALIAS | FINGERPRINT  | PUBLIC |                 DESCRIPTION                 |  ARCH  |   SIZE   |         UPLOAD DATE          |
+-------+--------------+--------+---------------------------------------------+--------+----------+------------------------------+
|       | 52ad4dc896ba | no     | ubuntu 16.04 LTS amd64 (release) (20180622) | x86_64 | 157.54MB | Jun 23, 2018 at 7:17am (UTC) |
+-------+--------------+--------+---------------------------------------------+--------+----------+------------------------------+
gaynor@gaynor-Aspire-E5-521:~$ lxc list
+-----------------+---------+-----------------------+------+------------+-----------+
|      NAME       |  STATE  |         IPV4          | IPV6 |    TYPE    | SNAPSHOTS |
+-----------------+---------+-----------------------+------+------------+-----------+
| informed-impala | RUNNING | 10.126.26.135 (eth0)  |      | PERSISTENT | 0         |
|                 |         | 10.124.253.1 (lxdbr0) |      |            |           |
+-----------------+---------+-----------------------+------+------------+-----------+

gaynor@gaynor-Aspire-E5-521:~$ lxc config set informed-impala security.nesting true
gaynor@gaynor-Aspire-E5-521:~$ lxc config get informed-impala security.nesting
true

gaynor@gaynor-Aspire-E5-521:~$ lxc exec informed-impala bash

root@informed-impala:~# lxc image list
+-------+--------------+--------+------------------------------------------+--------+----------+-------------------------------+
| ALIAS | FINGERPRINT  | PUBLIC |               DESCRIPTION                |  ARCH  |   SIZE   |          UPLOAD DATE          |
+-------+--------------+--------+------------------------------------------+--------+----------+-------------------------------+
|       | 9ff0350ad5c3 | no     | Archlinux current amd64 (20180624_01:27) | x86_64 | 134.16MB | Jun 24, 2018 at 11:24am (UTC) |
+-------+--------------+--------+------------------------------------------+--------+----------+-------------------------------+

root@informed-impala:~# lxc launch 9ff0350ad5c3
Creating the container
Container name is: nice-malamute
Starting nice-malamute
error: Error calling 'lxd forkstart nice-malamute /var/lib/lxd/containers /var/log/lxd/nice-malamute/lxc.conf': err='Failed to run: /usr/bin/lxd forkstart nice-malamute /var/lib/lxd/containers /var/log/lxd/nice-malamute/lxc.conf: '
  lxc 20180624125231.636 ERROR lxc_utils - utils.c:safe_mount:1739 - Operation not permitted - Failed to mount proc onto /usr/lib/x86_64-linux-gnu/lxc/proc
  lxc 20180624125231.636 ERROR lxc_conf - conf.c:lxc_mount_auto_mounts:734 - Operation not permitted - error mounting proc on /usr/lib/x86_64-linux-gnu/lxc/proc flags 14
  lxc 20180624125231.636 ERROR lxc_conf - conf.c:lxc_setup:4008 - failed to setup the automatic mounts for 'nice-malamute'
  lxc 20180624125231.636 ERROR lxc_start - start.c:do_start:811 - Failed to setup container "nice-malamute".
  lxc 20180624125231.636 ERROR lxc_sync - sync.c:__sync_wait:57 - An error occurred in another process (expected sequence number 3)
  lxc 20180624125231.695 ERROR lxc_start - start.c:__lxc_start:1358 - Failed to spawn container "nice-malamute".
  lxc 20180624125232.287 ERROR lxc_conf - conf.c:run_buffer:416 - Script exited with status 1.
  lxc 20180624125232.287 ERROR lxc_start - start.c:lxc_fini:546 - Failed to run lxc.hook.post-stop for container "nice-malamute".

Try `lxc info --show-log local:nice-malamute` for more info
  
root@informed-impala:~# lxc info --show-log local:nice-malamute
Name: nice-malamute
Remote: unix://
Architecture: x86_64
Created: 2018/06/24 12:52 UTC
Status: Stopped
Type: persistent
Profiles: default

Log:

            lxc 20180624125231.636 ERROR    lxc_utils - utils.c:safe_mount:1739 - Operation not permitted - Failed to mount proc onto /usr/lib/x86_64-linux-gnu/lxc/proc
            lxc 20180624125231.636 ERROR    lxc_conf - conf.c:lxc_mount_auto_mounts:734 - Operation not permitted - error mounting proc on /usr/lib/x86_64-linux-gnu/lxc/proc flags 14
            lxc 20180624125231.636 ERROR    lxc_conf - conf.c:lxc_setup:4008 - failed to setup the automatic mounts for 'nice-malamute'
            lxc 20180624125231.636 ERROR    lxc_start - start.c:do_start:811 - Failed to setup container "nice-malamute".
            lxc 20180624125231.636 ERROR    lxc_sync - sync.c:__sync_wait:57 - An error occurred in another process (expected sequence number 3)
            lxc 20180624125231.695 ERROR    lxc_start - start.c:__lxc_start:1358 - Failed to spawn container "nice-malamute".
            lxc 20180624125232.287 ERROR    lxc_conf - conf.c:run_buffer:416 - Script exited with status 1.
            lxc 20180624125232.287 ERROR    lxc_start - start.c:lxc_fini:546 - Failed to run lxc.hook.post-stop for container "nice-malamute".
            lxc 20180624125232.287 WARN     lxc_commands - commands.c:lxc_cmd_rsp_recv:177 - Command get_cgroup failed to receive response: Connection reset by peer.
            lxc 20180624125232.287 WARN     lxc_commands - commands.c:lxc_cmd_rsp_recv:177 - Command get_cgroup failed to receive response: Connection reset by peer.

# Just in case it's useful...

gaynor@gaynor-Aspire-E5-521:~$ lxc info --show-log local:informed-impala
Name: informed-impala
Remote: unix://
Architecture: x86_64
Created: 2018/06/23 07:17 UTC
Status: Running
Type: persistent
Profiles: default
Pid: 24663
Ips:
  eth0:	inet	10.126.26.135	vethQ0OXJF
  eth0:	inet6	fe80::216:3eff:fe10:d05f	vethQ0OXJF
  lo:	inet	127.0.0.1
  lo:	inet6	::1
  lxdbr0:	inet	10.124.253.1
  lxdbr0:	inet6	fe80::301f:c1ff:fe8e:5c9c
Resources:
  Processes: 43
  CPU usage:
    CPU usage (in seconds): 174
  Memory usage:
    Memory (current): 980.82MB
    Memory (peak): 1.43GB
  Network usage:
    eth0:
      Bytes received: 167.45MB
      Bytes sent: 3.98MB
      Packets received: 118562
      Packets sent: 61183
    lo:
      Bytes received: 0B
      Bytes sent: 0B
      Packets received: 0
      Packets sent: 0
    lxdbr0:
      Bytes received: 0B
      Bytes sent: 570B
      Packets received: 0
      Packets sent: 7

Log:

lxc informed-impala 20180623071740.994 WARN     lxc_conf - conf.c:lxc_map_ids:2855 - newuidmap binary is missing
lxc informed-impala 20180623071740.995 WARN     lxc_conf - conf.c:lxc_map_ids:2861 - newgidmap binary is missing
lxc informed-impala 20180623071741.165 WARN     lxc_conf - conf.c:lxc_map_ids:2855 - newuidmap binary is missing
lxc informed-impala 20180623071741.166 WARN     lxc_conf - conf.c:lxc_map_ids:2861 - newgidmap binary is missing

+
Some basic extra information


root@informed-impala:~# lxc image list
+-------+--------------+--------+------------------------------------------+--------+----------+-------------------------------+
| ALIAS | FINGERPRINT  | PUBLIC |               DESCRIPTION                |  ARCH  |   SIZE   |          UPLOAD DATE          |
+-------+--------------+--------+------------------------------------------+--------+----------+-------------------------------+
|       | 9ff0350ad5c3 | no     | Archlinux current amd64 (20180624_01:27) | x86_64 | 134.16MB | Jun 24, 2018 at 11:24am (UTC) |
+-------+--------------+--------+------------------------------------------+--------+----------+-------------------------------+

root@informed-impala:~# lxc list
+---------------+---------+------+------+------------+-----------+
|     NAME      |  STATE  | IPV4 | IPV6 |    TYPE    | SNAPSHOTS |
+---------------+---------+------+------+------------+-----------+
| nice-malamute | STOPPED |      |      | PERSISTENT | 0         |
+---------------+---------+------+------+------------+-----------+
root@informed-impala:~# lxc info nice-malamute
Name: nice-malamute
Remote: unix://
Architecture: x86_64
Created: 2018/06/24 12:52 UTC
Status: Stopped
Type: persistent
Profiles: default
			
<<<Console Sequence Capture>>>		   

# Information to attach

 - [x] Any relevant kernel output (`dmesg`)
   Kernel dmesg output attached.

[top_level_kernel.txt](https://github.com/lxc/lxd/files/2130931/top_level_kernel.txt)

 
 - [x] Container log (`lxc info NAME --show-log`)
   The "Outer container" log and "Inner container" log are both included in the attached information above.

 - [x] Container configuration (`lxc config show NAME --expanded`)
   -----------------------------------------------------------------------------------------------------------------------------------------------
   Outer Container ( informed-impala [Xenial]):
   -----------------------------------------------------------------------------------------------------------------------------------------------
   gaynor@gaynor-Aspire-E5-521:~$ lxc config show informed-impala --expanded
   architecture: x86_64
   config:
     image.architecture: amd64
     image.description: ubuntu 16.04 LTS amd64 (release) (20180622)
     image.label: release
     image.os: ubuntu
     image.release: xenial
     image.serial: "20180622"
     image.version: "16.04"
     security.nesting: "true"
     volatile.base_image: 52ad4dc896ba051f5efd9ced66dd4088167271febee4ae203eb6d9ccb39a9916
     volatile.eth0.hwaddr: 00:16:3e:10:d0:5f
     volatile.idmap.base: "0"
     volatile.idmap.next: '[{"Isuid":true,"Isgid":true,"Hostid":1000000,"Nsid":0,"Maprange":1000000000}]'
     volatile.last_state.idmap: '[{"Isuid":true,"Isgid":true,"Hostid":1000000,"Nsid":0,"Maprange":1000000000}]'
     volatile.last_state.power: RUNNING
   devices:
     eth0:
       name: eth0
       nictype: bridged
       parent: lxdbr0
       type: nic
     root:
       path: /
       pool: default
       type: disk
   ephemeral: false
   profiles:
   - default
   stateful: false
   description: ""

   -----------------------------------------------------------------------------------------------------------------------------------------------
   Innere Container ( nice-malamute [ArchLinux]]):  **** ERROR *** Thrown!!!
   -----------------------------------------------------------------------------------------------------------------------------------------------
   
   root@informed-impala:~# lxc list
   +---------------+---------+------+------+------------+-----------+
   |     NAME      |  STATE  | IPV4 | IPV6 |    TYPE    | SNAPSHOTS |
   +---------------+---------+------+------+------------+-----------+
   | nice-malamute | STOPPED |      |      | PERSISTENT | 0         |
   +---------------+---------+------+------+------------+-----------+

   root@informed-impala:~# lxc config show nice-malmute --expanded
   error: not found

   root@informed-impala:~# lxc config show local:nice-malmute --expanded
   error: not found

   What the blazes !!! ???
   -----------------------------------------------------------------------------------------------------------------------------------------------
 
 - [x] Main daemon log (at /var/log/lxd/lxd.log or /var/snap/lxd/common/lxd/logs/lxd.log)

   Log files found from Yakkety Host PC:
   -------------------------------
   /var/snap/lxd/common/lxd/logs/lxd.log 
   
[var_snap_lxd_common_lxd_logs_lxd.log](https://github.com/lxc/lxd/files/2130935/var_snap_lxd_common_lxd_logs_lxd.log)

   /var/snap/lxd/common/lxd/logs/informed-impala/console.log   
  
[var_snap_lxd_common_lxd_logs_informed-impala_console.log](https://github.com/lxc/lxd/files/2130936/var_snap_lxd_common_lxd_logs_informed-impala_console.log)

   /var/snap/lxd/common/lxd/logs/informed-impala/lxc.log   

[var_snap_lxd_common_lxd_logs_informed-impala_lxc.log](https://github.com/lxc/lxd/files/2130937/var_snap_lxd_common_lxd_logs_informed-impala_lxc.log)


   Log files found from (outer) container "informed-impala" (Xenial)
   -----------------------------------------------------------------
   /var/log/lxd/lxd.log
   
[var_log_lxd_lxd.log](https://github.com/lxc/lxd/files/2130941/var_log_lxd_lxd.log)

   /var/log/lxd/nice-malamute/lxc.log

[var_log_lxd_nice-malamute_lxc.log](https://github.com/lxc/lxd/files/2130942/var_log_lxd_nice-malamute_lxc.log)

 
 - [x] Output of the client with --debug
   From inside the (outer) container "informed-impala" (Xenial), tried to start up the (inner) container ""

   root@informed-impala:~# lxc list
+---------------+---------+------+------+------------+-----------+
|     NAME      |  STATE  | IPV4 | IPV6 |    TYPE    | SNAPSHOTS |
+---------------+---------+------+------+------------+-----------+
| nice-malamute | STOPPED |      |      | PERSISTENT | 0         |
+---------------+---------+------+------+------------+-----------+
root@informed-impala:~# lxc start nice-malamute --debug
DBUG[06-24|14:43:46] Connecting to a local LXD over a Unix socket 
DBUG[06-24|14:43:46] Sending request to LXD                   etag= method=GET url=http://unix.socket/1.0
DBUG[06-24|14:43:46] Got response struct from LXD 
DBUG[06-24|14:43:46] 
	{
		"config": {},
		"api_extensions": [
			"id_map",
			"id_map_base",
			"resource_limits"
		],
		"api_status": "stable",
		"api_version": "1.0",
		"auth": "trusted",
		"auth_methods": null,
		"public": false,
		"environment": {
			"addresses": [],
			"architectures": [
				"x86_64",
				"i686"
			],
			"certificate": "-----BEGIN CERTIFICATE-----\nMIIFWzCCA0OgAwIBAgIQHtrhM8QUxV7OWT9+Mg3osjANBgkqhkiG9w0BAQsFADA9\nMRwwGgYDVQQKExNsaW51eGNvbnRhaW5lcnMub3JnMR0wGwYDVQQDDBRyb290QGlu\nZm9ybWVkLWltcGFsYTAeFw0xODA2MjQxMTA4MTFaFw0yODA2MjExMTA4MTFaMD0x\nHDAaBgNVBAoTE2xpbnV4Y29udGFpbmVycy5vcmcxHTAbBgNVBAMMFHJvb3RAaW5m\nb3JtZWQtaW1wYWxhMIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEArYYc\nZngLN8JQMzb1j2qH93uBdtpJWZC85+ja5M0yB6Onm4AwRnTKhyNlfQ39xv5/ffm3\nqXhPyBdOfWycoD6NO/a7G487CiwOUPkhwEH5zp04sCclzC5xpwKPPsQSFytux2/x\nEd1a08Moiw5antRKOKG3jFoO72XndLwF28LEx1/eHZBltaqjQcPOZsQc9kgp4r6V\nDYNM5hLdRwnD9ZQniNB7hrIW+pq0YGo2G1Gm4VwmQLdAPwwF/tDkqhyw3JPcTpbs\nwapy7chun3VzwJk6dNqMaEx8b935QLA7+WBqrdr8WuwdftP8gQE8qaTOrlSILyLC\nVbElPpR8CE9Q5UGJP6R7z/bG/39cJz/HE6fatXTQpB8VT93b7LP/3JHObTrCv69h\niNG2SEgwp9pryrZH5nAw6+pWG00LBT+hdrfUfC7TfHYHt0bQ9DB3RZdgz+fXKo6b\n/fHvcbIcCcixND1Q7qlaCAtKGH+kuudfIikmSbdXsdiB5eg788+NSYrFXUbUSM0c\n/Qoemm8PIMXOQyxg+3A1mXyTLoMNHhhqWlZHrZiKqQ2lvW0RRrMEPesai+m4zUxP\nofPQWaIrCOAV8KKIjOfI/9/tEYptoF81j/9GpFx7+sUUYhKXZGsrJGW7yJKB+SxX\nVjj1TnsRW1xXDR/z0xaxDcM+fm735/SqBTf+IqECAwEAAaNXMFUwDgYDVR0PAQH/\nBAQDAgWgMBMGA1UdJQQMMAoGCCsGAQUFBwMBMAwGA1UdEwEB/wQCMAAwIAYDVR0R\nBBkwF4IPaW5mb3JtZWQtaW1wYWxhhwQKfhqHMA0GCSqGSIb3DQEBCwUAA4ICAQBO\nE5OtGlQ8gGUAfVKeNsa/Z759Jxvwn5fXN3bqIxTThtpGNbbK05CRVFRduknH+fYU\nXzukwPou6L9b2l2QHxTWPF7WxXbBOVjy8dRffZFY4sy+8aMgQPWVCYGRJPG5JwYc\nb4z6HpISPD8nL1vy69Xsz/QnYvqQLrNMrsuuhAn1i1r9GJC2CxMtNmOEQJ9SmNzh\ns6bR80T882apqE7u0HOXmJj+pMkh99yPBtfL+EyRmMa31a7W0tYHKdNNntPFwn0g\n1aAJnbFM1OEB/ppARyifHdFTpt8u5bcsLDnpkx98/bZEPnTsu6fwiKciLsJLTl2Q\n1loLq06OSqB+L2gB64cn1qqSUbSwJOgWTVIna/uAzmKKFXBOvXwjx+UlQM63/yb3\nM/azST7kBpbTF0TjtekbOSPqYfbYKaorzwT2SVMcL2qjT1xPIIrh+KwIOEzh1hpZ\nlkYeuIzKD0AvC8VUkYhInGyDRtcJhquHHsiQjJY5BF+JgcovjGSoEluG+UgcMkh5\nPaGcHrddRblYngosuRG0PP6uDb4LbzwTZT7eenThWnzHHgN2RupNSfR5tliqpEZh\nQ7U/9HpmdUDMDu7cOXfHU73yHhtQ1saEakeAO8cjVeJ3BxIw+euQbe3ISnGsBo7X\nIzfvpX4q9nCBgnbMujkGIwzBnOzlJgnNDCuM1GVWBQ==\n-----END CERTIFICATE-----\n",
			"certificate_fingerprint": "b2171082932ac6fce9df4997bf43d94a77c73ae8c852e1a4cf835c0f05857915",
			"driver": "lxc",
			"driver_version": "2.0.8",
			"kernel": "Linux",
			"kernel_architecture": "x86_64",
			"kernel_version": "4.8.0-59-generic",
			"server": "lxd",
			"server_pid": 1672,
			"server_version": "2.0.11",
			"storage": "dir",
			"storage_version": ""
		}
	} 
DBUG[06-24|14:43:46] Sending request to LXD                   etag= method=GET url=http://unix.socket/1.0/containers/nice-malamute
DBUG[06-24|14:43:46] Got response struct from LXD 
DBUG[06-24|14:43:46] 
	{
		"architecture": "x86_64",
		"config": {
			"volatile.base_image": "9ff0350ad5c3bedf9b2ac348ce67da159cf362ad13af4a68a8b3d04a9c9f3617",
			"volatile.eth0.hwaddr": "00:16:3e:57:06:49",
			"volatile.idmap.base": "0",
			"volatile.idmap.next": "[{\"Isuid\":true,\"Isgid\":false,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":65536},{\"Isuid\":false,\"Isgid\":true,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":65536}]",
			"volatile.last_state.idmap": "[{\"Isuid\":true,\"Isgid\":false,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":65536},{\"Isuid\":false,\"Isgid\":true,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":65536}]",
			"volatile.last_state.power": "RUNNING"
		},
		"devices": {
			"root": {
				"path": "/",
				"type": "disk"
			}
		},
		"ephemeral": false,
		"profiles": [
			"default"
		],
		"stateful": false,
		"description": "",
		"created_at": "2018-06-24T12:52:20Z",
		"expanded_config": {
			"environment.http_proxy": "",
			"user.network_mode": "",
			"volatile.base_image": "9ff0350ad5c3bedf9b2ac348ce67da159cf362ad13af4a68a8b3d04a9c9f3617",
			"volatile.eth0.hwaddr": "00:16:3e:57:06:49",
			"volatile.idmap.base": "0",
			"volatile.idmap.next": "[{\"Isuid\":true,\"Isgid\":false,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":65536},{\"Isuid\":false,\"Isgid\":true,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":65536}]",
			"volatile.last_state.idmap": "[{\"Isuid\":true,\"Isgid\":false,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":65536},{\"Isuid\":false,\"Isgid\":true,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":65536}]",
			"volatile.last_state.power": "RUNNING"
		},
		"expanded_devices": {
			"eth0": {
				"name": "eth0",
				"nictype": "bridged",
				"parent": "lxdbr0",
				"type": "nic"
			},
			"root": {
				"path": "/",
				"type": "disk"
			}
		},
		"name": "nice-malamute",
		"status": "Stopped",
		"status_code": 102,
		"last_used_at": "0001-01-01T00:00:00Z"
	} 
DBUG[06-24|14:43:46] Connected to the websocket 
DBUG[06-24|14:43:46] Sending request to LXD                   etag= method=PUT url=http://unix.socket/1.0/containers/nice-malamute/state
DBUG[06-24|14:43:46] 
	{
		"action": "start",
		"timeout": 0,
		"force": false,
		"stateful": false
	} 
DBUG[06-24|14:43:46] Got operation from LXD 
DBUG[06-24|14:43:46] 
	{
		"id": "d36a43e5-88c3-4ba5-bdc1-fbfc6b8baff4",
		"class": "task",
		"created_at": "2018-06-24T14:43:46.788019549Z",
		"updated_at": "2018-06-24T14:43:46.788019549Z",
		"status": "Running",
		"status_code": 103,
		"resources": {
			"containers": [
				"/1.0/containers/nice-malamute"
			]
		},
		"metadata": null,
		"may_cancel": false,
		"err": ""
	} 
DBUG[06-24|14:43:46] Sending request to LXD                   etag= method=GET url=http://unix.socket/1.0/operations/d36a43e5-88c3-4ba5-bdc1-fbfc6b8baff4
DBUG[06-24|14:43:46] Got response struct from LXD 
DBUG[06-24|14:43:46] 
	{
		"id": "d36a43e5-88c3-4ba5-bdc1-fbfc6b8baff4",
		"class": "task",
		"created_at": "2018-06-24T14:43:46.788019549Z",
		"updated_at": "2018-06-24T14:43:46.788019549Z",
		"status": "Running",
		"status_code": 103,
		"resources": {
			"containers": [
				"/1.0/containers/nice-malamute"
			]
		},
		"metadata": null,
		"may_cancel": false,
		"err": ""
	} 
error: Error calling 'lxd forkstart nice-malamute /var/lib/lxd/containers /var/log/lxd/nice-malamute/lxc.conf': err='Failed to run: /usr/bin/lxd forkstart nice-malamute /var/lib/lxd/containers /var/log/lxd/nice-malamute/lxc.conf: '
  lxc 20180624144347.180 ERROR lxc_utils - utils.c:safe_mount:1739 - Operation not permitted - Failed to mount proc onto /usr/lib/x86_64-linux-gnu/lxc/proc
  lxc 20180624144347.180 ERROR lxc_conf - conf.c:lxc_mount_auto_mounts:734 - Operation not permitted - error mounting proc on /usr/lib/x86_64-linux-gnu/lxc/proc flags 14
  lxc 20180624144347.180 ERROR lxc_conf - conf.c:lxc_setup:4008 - failed to setup the automatic mounts for 'nice-malamute'
  lxc 20180624144347.180 ERROR lxc_start - start.c:do_start:811 - Failed to setup container "nice-malamute".
  lxc 20180624144347.180 ERROR lxc_sync - sync.c:__sync_wait:57 - An error occurred in another process (expected sequence number 3)
  lxc 20180624144347.239 ERROR lxc_start - start.c:__lxc_start:1358 - Failed to spawn container "nice-malamute".
  lxc 20180624144347.841 ERROR lxc_conf - conf.c:run_buffer:416 - Script exited with status 1.
  lxc 20180624144347.841 ERROR lxc_start - start.c:lxc_fini:546 - Failed to run lxc.hook.post-stop for container "nice-malamute".

Try `lxc info --show-log nice-malamute` for more info

 - [x] Output of the daemon with --debug (alternatively output of `lxc monitor` while reproducing the issue)
   CG: Sorry, I am not fully clear what you require me to do here !
 
 # Additional Information
 
 Directory structure for logs and noticed one broken symbolic link inside the (outer) container 'informed-impala'

``` ![selection_017](https://user-images.githubusercontent.com/5797613/41820595-d3033b82-77dc-11e8-8967-e581d1055d31.png)
![selection_016](https://user-images.githubusercontent.com/5797613/41820597-da0e77e8-77dc-11e8-8136-748fbc742333.png)

**Comment by stgraber** [comment]
Did you set security.nesting=true on the outer container, followed by a restart of the outer container?

**Comment by ColumGaynor** [comment]
CG: Wow that was fast... I did set the security.nesting = true on the outer container, but I did not realise
        that a restart of the outer container is required!   I will try that and report back immediately :smiley:

**Comment by ColumGaynor** [comment]
10 out of 10 sir! Fully correct - after I restarted the outer container, I can now start-up the inner container.
God Damn Amazing Stuff and Fifty Lashes to me for trying such a convoluted experiment. No doubt I could now launch a Docker Container from inside the inner-ArchLinux container :beer: 

One small thing I noticed ... the tab completion on container names does not work inside the outer container...  but it's small potatoes.

Is there any book / white paper explaining the detailed architecture of your LXD feature. Really nice concept.
